File: src/main/java/dev/araopj/hrplatformapi/employee/model/Employee.java
package dev.araopj.hrplatformapi.employee.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import dev.araopj.hrplatformapi.utils.EntityTimestamp;
import dev.araopj.hrplatformapi.utils.annotations.Uuid;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class Employee extends EntityTimestamp implements Serializable {

    @Id
    @Uuid
    private String id;

    @Column(unique = true, nullable = false)
    private Long employeeNumber;

    @Column(nullable = false)
    private String itemNumber;

    @Column(nullable = false)
    private String firstName;

    @Column
    private String middleName;

    @Column(nullable = false)
    private String lastName;

    @Column
    private String photo;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(nullable = false, unique = true)
    private String email;

    @Column
    private String phoneNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Gender gender;

    @Column(nullable = false)
    private String taxPayerIdentificationNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CivilStatus civilStatus;

    @Column
    private String bankAccountNumber;

    @Column
    boolean archived;

    @Column
    private String userId;

    @JsonIgnore
    @JsonManagedReference
    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<IdDocument> idDocuments;

    @JsonIgnore
    @JsonManagedReference
    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL)
    private Set<EmploymentInformation> employmentInformation;
}

File: frontend/src/lib/types/employee.ts
export type Gender = 'MALE' | 'FEMALE' | 'OTHER';

export type CivilStatus = 'SINGLE' | 'MARRIED' | 'WIDOWED';

export interface Employee {
	id: string;
	employeeNumber: string;
	itemNumber: string;
	firstName: string;
	middleName: string | null;
	lastName: string;
	photo: string | null;
	dateOfBirth: string;
	email: string;
	phoneNumber: string | null;
	gender: Gender;
	taxPayerIdentificationNumber: string;
	civilStatus: CivilStatus;
	bankAccountNumber: string | null;
	archived: boolean;
	userId: string | null;
	createdAt: string;
	updatedAt: string;
}

export interface EmployeeCreateRequest {
	employeeNumber: string;
	itemNumber: string;
	firstName: string;
	middleName?: string;
	lastName: string;
	photo?: string;
	dateOfBirth: string;
	email: string;
	phoneNumber: string;
	gender: Gender;
	taxPayerIdentificationNumber: string;
	civilStatus: CivilStatus;
	bankAccountNumber?: string;
}

export interface EmployeeUpdateRequest extends EmployeeCreateRequest {
	id: string;
}

export interface Pagination {
	page: number;
	size: number;
	totalElements: number;
	totalPages: number;
}

export interface ApiResponse<T> {
	timestamp: string;
	data: T;
	pagination?: Pagination;
}

export interface EmployeeListResponse {
	timestamp: string;
	data: Employee[];
	pagination: Pagination;
}
